### PR TITLE
Logger

### PR DESCRIPTION
**Pull request by ricleal**
Logger Changed. Now we have configuration files: `logging.ini`

### CODE REVIEW

**Comment by ricleal** [comment]
#fixes #916

**Comment by ricleal** [comment]
Sure! I can change on runtime all loggers to debug if a `DEBUG` flag or whatever is set.

**Comment by pkienzle** [comment]
Somehow set it to run from "python run.py", which is only used by developers.  Note that Andrew merged my patch which cloned the logging output to the console when run as ./run.py; I don't know if that will conflict.  There is a ticket to close when this patch is complete.  Don't know the number.

**Comment by ajj** [comment]
I'm still getting double logging to the console even if I back out the changes from @pkienzle PR #58

**Comment by ricleal** [comment]
Double logging solved!

**Comment by ricleal** [comment]
```
----------------------------------------------
Results by test modules:
    PASSED: 15
    FAILED: 0    (0%)
Results by tests:
    Tests run:    147
    Tests failed: 0
    Test errors:  0
----------------------------------------------
```

**Comment by andyfaff** [comment]
The test failure on travis is:

```
+/home/travis/miniconda/bin/python utest_sasview.py
/home/travis/build/SasView/sasview/test/logging.ini
Traceback (most recent call last):
  File "utest_sasview.py", line 11, in <module>
    logging.config.fileConfig(LOGGER_CONFIG_FILE)
  File "/home/travis/miniconda/lib/python2.7/logging/config.py", line 85, in fileConfig
    handlers = _install_handlers(cp, formatters)
  File "/home/travis/miniconda/lib/python2.7/logging/config.py", line 163, in _install_handlers
    h = klass(*args)
  File "/home/travis/miniconda/lib/python2.7/logging/__init__.py", line 913, in __init__
    StreamHandler.__init__(self, self._open())
  File "/home/travis/miniconda/lib/python2.7/logging/__init__.py", line 943, in _open
    stream = open(self.baseFilename, self.mode)
IOError: [Errno 2] No such file or directory: '/home/travis/.sasview/tests.log
```

**Comment by andyfaff** [comment]
So in that circumstance the logger is trying to write to `tests.log` in the `~/.sasview` directory. What creates that directory and when? Because the travis tests are done on a clean container each time `~/.sasview` won't exist when `utest_sasview.py` is run.

File: test/utest_sasview.py [hunk 1]
```diff
@@ -3,11 +3,19 @@
 import subprocess
 import re
 import sys
+
+import logging
+import logging.config
+LOGGER_CONFIG_FILE = os.path.join(os.path.abspath(os.path.dirname(__file__)), 'logging.ini')
+print LOGGER_CONFIG_FILE
```
Comment (andyfaff) [inline test/utest_sasview.py]: `print` usage should be as a function, not a statement. i.e. `print(LOGGER_CONFIG_FILE`.

**Comment by ricleal** [comment]
Removed the print. Problem is here:
`tests/logger.ini`
```ini
[handler_log_file]
class=logging.FileHandler
level=INFO
formatter=detailed
args=(os.path.join(os.path.expanduser("~"),'.sasview/tests.log'),"w")
```
Will see if there's an option to create inexistent directories.

**Comment by andyfaff** [comment]
Indeed, `utest_sasview.py` fails on my computer when the `~/.sasview` directory is not present. It then works when I create it.

**Comment by andyfaff** [comment]
What is the `.sasview` directory used for? Perhaps it should be created when sasview is installed, when it's run for the first time, or when the tests are run?

**Comment by ricleal** [comment]
`.sasview` is created by `write_batch_tofile(self, data, file_name, details="")` in `gui_manager.py`.

The log file is just created in `.sasview`. No info is written to it. I will see if it's really needed... There was some hardcoded stuff writing to a file before...

**Comment by ajj** [comment]
.sasview holds user plugins, local configurations, and custom categories. Probably shouldn't be needed to run tests...

**Pull request merged**